From 20d74a29c1c9050880f0456b2c3a0242c9c8ab70 Mon Sep 17 00:00:00 2001
From: Sasi Kumar Maddineni <quic_sasikuma@quicinc.com>
Date: Thu, 5 Jun 2025 15:02:40 +0530
Subject: [PATCH] Mount /run with tmpfs

Upstream-Status: Pending

Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
Git-Commit: 0c3b2340811774b495920c397c5603448446d160
Git-repo: https://github.com/foundriesio/meta-lmp

[Sasi Kumar Maddineni <quic_sasikuma@quicinc.com>: Resolved merge conflicts]
Signed-off-by: Sasi Kumar Maddineni <quic_sasikuma@quicinc.com>
---
 finish | 3 ++-
 init   | 4 +++-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/finish b/finish
index ac0de9f..ed8298d 100755
--- a/finish
+++ b/finish
@@ -35,10 +35,11 @@ finish_run() {
 			mount -n --move "$dir" "${ROOTFS_DIR}/media/${dir##*/}"
 		done
 
-		debug "Moving /dev, /proc and /sys onto rootfs..."
+		debug "Moving /dev, /proc, /sys and /run onto rootfs..."
 		mount --move /dev $ROOTFS_DIR/dev
 		mount --move /proc $ROOTFS_DIR/proc
 		mount --move /sys $ROOTFS_DIR/sys
+		mount --move /run $ROOTFS_DIR/run
 
 		cd $ROOTFS_DIR
 		exec switch_root -c /dev/console $ROOTFS_DIR ${bootparam_init:-/sbin/init}
diff --git a/init b/init
index 20c0455..495d157 100755
--- a/init
+++ b/init
@@ -78,9 +78,11 @@ EFI_DIR=/sys/firmware/efi  # place to store device firmware information
 touch /etc/fstab
 
 # initialize /proc, /sys, /run/lock and /var/lock
-mkdir -p /proc /sys /run/lock /var/lock
+mkdir -p /proc /sys /run
 mount -t proc proc /proc
 mount -t sysfs sysfs /sys
+mount -t tmpfs tmpfs /run
+mkdir -p /run/lock /var/lock
 
 if [ -d $EFI_DIR ];then
 	mount -t efivarfs none /sys/firmware/efi/efivars
-- 
2.34.1

